### PR TITLE
Reformat / Refactor of states in the discover Viewmodel

### PR DESCRIPTION
**Pull request by JeremyHugentobler**
## The issue
Previously, all the states and functions of the discoverscreen viewmodel were given as parameters to the screens using them. This made the code hard to read and maintain and had a negative impact on the general line coverage of the project. (Because the navgraph is almost untestable, adding lines there to initiate screens wasn't beneficial for us)

## The solution
I fixed this for the discovery screen viewmodel by creating two data classes, one that stores all the states and one that stores all the functions called by screens. This took a lot of time to merge with main since a lot of PRs changed completely the content of certain files.

## Tests
I tested as much as it was possible, but Sonar has a weird way of qualifying what is "new code". It counted in my additions a lot of part that I didn't really modify (the map screen) that blocks my line coverage at around 73. There is for the moment no possible ways to test this composable and since I did not change anything about it's functionalities, the PR will sadly not achieve 80%.

## Next
We should do that for all the viewmodel, but this time is really time consuming and comports some risks when merging

### CODE REVIEW

**Review by ecornamu — APPROVED** [review]
LGTM

File: app/src/main/java/com/lastaoutdoor/lasta/ui/navigation/MainNavGraph.kt [hunk 1]
```diff
@@ -40,71 +42,38 @@ import com.lastaoutdoor.lasta.viewmodel.WeatherViewModel
...
 
       PermissionManager(discoverScreenViewModel::updatePermission)
 
       DiscoverScreen(
-          isLoading,
-          activities,
-          screen,
-          range,
-          centerPoint,
+          discoverScreenState,
```
Comment (Andy130604) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/navigation/MainNavGraph.kt:64]: This is so nice to have so few arguments!

File: app/src/main/java/com/lastaoutdoor/lasta/ui/navigation/MainNavGraph.kt [hunk 2]
```diff
@@ -187,59 +161,50 @@ fun NavGraphBuilder.addMainNavGraph(navController: NavHostController) {
...
-          initialZoom,
-          activities,
-          discoverScreenViewModel::updateActivities,
-          discoverScreenViewModel::updateMarkers,
-          discoverScreenViewModel::updateSelectedMarker,
-          discoverScreenViewModel::clearSelectedItinerary,
-          discoverScreenViewModel::fetchActivities,
-          selectedZoom,
+          discoverScreenState,
+          discoverScreenCallBacks,
```
Comment (Andy130604) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/navigation/MainNavGraph.kt:183]: You did also for this, this is really nice, I love it!

File: app/src/main/java/com/lastaoutdoor/lasta/viewmodel/DiscoverScreenViewModel.kt [hunk 3]
```diff
@@ -34,6 +34,60 @@ import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.tasks.await
 
+// Initial values
+const val INITIAL_RANGE = 10000.0
+
+// All the function of the viewmodel that the view can call
+data class DiscoverScreenCallBacks(
```
Comment (Andy130604) [inline app/src/main/java/com/lastaoutdoor/lasta/viewmodel/DiscoverScreenViewModel.kt:41]: Everything in one data class, so easy to maintain for the future

**Review by Andy130604 — APPROVED** [review]
You did some amazing changes! Maybe not so visually appealing but truly useful for everyone. Your coverage is not above 80% on new code but since you are using the map I guess it isn't testable. Everything else is tested Overall really good job. Keep it up :)

**Pull request merged**